pkg/tiller/release_uninstall.go

```diff
@@ -96,9 +96,24 @@ func (s *ReleaseServer) UninstallRelease(c ctx.Context, req *services.UninstallR
 		}
 	}
 
+	// if some resources delete failed, set status to UNKNOWN and return.
+	if len(es) > 0 {
+		rel.Info.Status.Code = release.Status_UNKNOWN
```

Comment:
This should not be set to UNKNOWN. It should be set to FAILED or perhaps a new FAILED_DELETE status should be set. UNKNOWN is reserved for special circumstances when no status can be determined. In this case, we know for sure that we're in a bad state.

Reply:
OK, I set to FAILED, firstly